File: Data Structures & Algorithms/car-fleet/submission-2.java
class Solution {
    public int carFleet(int target, int[] position, int[] speed) {
        int[][] cars = new int[position.length][2];
        for(int i=0; i<position.length; i++){
            cars[i][0] = position[i];
            cars[i][1] = speed[i];
        }
        Arrays.sort(cars, (a,b) -> Integer.compare(b[0], a[0])); //descending by position
        Deque<Double> stack = new ArrayDeque<>();
        for(int[] car: cars){
            double time = (double)(target-car[0])/car[1];
            if(stack.isEmpty() || time > stack.peek()){
                stack.push(time);
            }
        }
        return stack.size();
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        i,j = 0, len(numbers)-1
        while i<j:
            currSum = numbers[i] + numbers[j]

            if currSum > target:
                j -= 1
            elif currSum < target:
                i+=1
            else:
                return [i+1, j+1]
            
        return []